### PR TITLE
soften links

### PR DESCRIPTION
**Pull request by russelltsherman**

### CODE REVIEW

**Comment by russelltsherman** [comment]
now with less ENVVAR

File: _admin/art.php [hunk 1]
```diff
@@ -22,11 +22,11 @@
...
-            <a href="https://secure.burningflipside.com/register/api/v1/art?fmt=xml">Data Dump Of Doom (.xml)</a> |
-            <a href="https://secure.burningflipside.com/register/api/v1/art/*/artLead?fmt=csv">Art Leads (.csv)</a> |
-            <a href="https://secure.burningflipside.com/register/api/v1/art/*/soundLead?fmt=csv">Sound Leads (.csv)</a> |
-            <a href="https://secure.burningflipside.com/register/api/v1/art/*/safetyLead?fmt=csv">Safety Leads (.csv)</a> |
-            <a href="https://secure.burningflipside.com/register/api/v1/art/*/cleanupLead?fmt=csv">Cleanup Leads (.csv)</a>
+            <a href="'.$page->secureUrl.'/register/api/v1/art?fmt=xml">Data Dump Of Doom (.xml)</a> |
+            <a href="'.$page->secureUrl.'/register/api/v1/art/*/artLead?fmt=csv">Art Leads (.csv)</a> |
+            <a href="'.$page->secureUrl.'/register/api/v1/art/*/soundLead?fmt=csv">Sound Leads (.csv)</a> |
+            <a href="'.$page->secureUrl.'/register/api/v1/art/*/safetyLead?fmt=csv">Safety Leads (.csv)</a> |
+            <a href="'.$page->secureUrl.'/register/api/v1/art/*/cleanupLead?fmt=csv">Cleanup Leads (.csv)</a>
```
Comment (pboyd04) [inline _admin/art.php]: I'd rather just make these relative links (i.e. ../api). That seem like a good idea to you?

File: _admin/dmv.php [hunk 2]
```diff
@@ -22,7 +22,7 @@
             </div>
         </div>
         <div class="row">
-            <a href="https://secure.burningflipside.com/register/api/v1/dmv?fmt=xml">Data Dump Of Doom (.xml)</a>
+            <a href="'.$page->secureUrl.'/register/api/v1/dmv?fmt=xml">Data Dump Of Doom (.xml)</a>
```
Comment (pboyd04) [inline _admin/dmv.php]: Same comment, relative link seems like a better solution here.

File: class.RegisterWizardPage.php [hunk 3]
```diff
@@ -406,7 +406,7 @@ function print_page($header=true)
     <div id="content">
         <div class="row">
             <div class="col-lg-12">
-                <h1 class="page-header">You must <a href="https://profiles.burningflipside.com/login.php?return='.$this->currentURL().'">log in <span class="glyphicon glyphicon-log-in"></span></a> to access the Burning Flipside Registration system!</h1>
+                <h1 class="page-header">You must <a href="'.$this->profilesUrl.'/login.php?return='.$this->currentURL().'">log in <span class="glyphicon glyphicon-log-in"></span></a> to access the Burning Flipside Registration system!</h1>
```
Comment (pboyd04) [inline class.RegisterWizardPage.php]: $this->loginUrl

File: view_art.php [hunk 4]
```diff
@@ -12,7 +12,7 @@
     <div id="content">
         <div class="row">
             <div class="col-lg-12">
-                <h1 class="page-header">You must <a href="https://profiles.burningflipside.com/login.php?return='.$page->currentURL().'">log in <span class="glyphicon glyphicon-log-in"></span></a> to access the Burning Flipside Registration system!</h1>
+                <h1 class="page-header">You must <a href="'.$page->profilesUrl.'/login.php?return='.$page->currentURL().'">log in <span class="glyphicon glyphicon-log-in"></span></a> to access the Burning Flipside Registration system!</h1>
```
Comment (pboyd04) [inline view_art.php]: $this->loginUrl

File: view_dmv.php [hunk 5]
```diff
@@ -12,7 +12,7 @@
     <div id="content">
         <div class="row">
             <div class="col-lg-12">
-                <h1 class="page-header">You must <a href="https://profiles.burningflipside.com/login.php?return='.$page->currentURL().'">log in <span class="glyphicon glyphicon-log-in"></span></a> to access the Burning Flipside Registration system!</h1>
+                <h1 class="page-header">You must <a href="'.$page->profilesUrl.'/login.php?return='.$page->currentURL().'">log in <span class="glyphicon glyphicon-log-in"></span></a> to access the Burning Flipside Registration system!</h1>
```
Comment (pboyd04) [inline view_dmv.php]: $this->loginUrl

File: view_event.php [hunk 6]
```diff
@@ -12,7 +12,7 @@
     <div id="content">
         <div class="row">
             <div class="col-lg-12">
-                <h1 class="page-header">You must <a href="https://profiles.burningflipside.com/login.php?return='.$page->currentURL().'">log in <span class="glyphicon glyphicon-log-in"></span></a> to access the Burning Flipside Registration system!</h1>
+                <h1 class="page-header">You must <a href="'.$page->profilesUrl.'/login.php?return='.$page->currentURL().'">log in <span class="glyphicon glyphicon-log-in"></span></a> to access the Burning Flipside Registration system!</h1>
```
Comment (pboyd04) [inline view_event.php]: $this->loginUrl

**Comment by russelltsherman** [comment]
I'm not seeing a specific benefit to using loginUrl or logoutUrl
chances are these endpoints will never change through the entire lifespan of these applications
seems like these are variables which don't need to exist..
and we don't like those.. yes?

**Comment by pboyd04** [comment]
It's essentially the first version of your softening links changes. And since the FlipPage and FlipAdminPage need to know the login and logout URLs in order to render correctly this isn't unneeded coupling while, in my opinion, assuming they will always be part of profiles is. 

For example, if we chose to use Wordpress's session for SSO or MediaWiki's or some other future framework then the login and logout URL's would change to something other than profiles. 

What I don't like is coupling (i.e. interdependence of two modules). So if the Registration module needs to understand how login/logout works and what the structure of the profiles site is, that is what I don't like. Make sense?

**Comment by pboyd04** [comment]
Basically, the reason you have so many of these pulls all over the place is that currently the system is too tightly coupled. I'm ok with your changes, but it has pointed out a flaw that I think, long term, we need to address in a better way.

**Comment by russelltsherman** [comment]
if my uncle had a vagina he'd be my aunt... 
this statement makes about as much sense as keeping a variable around that we don't and will likely never need..

**Comment by pboyd04** [comment]
I'm really not following, we do need and use the variable today. How are you claiming we don't need it?

**Comment by russelltsherman** [comment]
loginUrl is always https://profiles.burningflipside.com/login.php
all system logins go through this single endpoint..
is that not true?

**Comment by pboyd04** [comment]
In production, yes. 

In your dev system it's localhost:<someport>/logout.php
In the gate system it's 10.0.0.1/logout.php (NOTE: NOT PROFILES!)
In my dev system it's 192.168.0.1/logout.php

**Comment by russelltsherman** [comment]
and in my local dev it's https://localhost:3300/login.php
the only part that ever changes is the FQDN / IP
$page->profilesUrl.'/login.php' is sufficient to satisfy ALL current use cases..
$page->loginUrl is an extra var to which there is no specific benefit..

**Comment by pboyd04** [comment]
Here's the thing. I'm ok with $this->profilesUrl to get you where you want to go. My intent is that it goes away.

Every page knowing where the login and logout scripts are make sense because we have every page render that link conditionally based on if there is a user logged in or not. Every page knowing where profiles is doesn't make as much sense, but I'm trying not to block your dev by insisting on perfect code purity (particularly when we don't have it to start with)

**Comment by pboyd04** [comment]
Let's get CommonCode resolved. Then I'm willing to accept this as is since my issues with it are minor ok?

**Comment by pboyd04** [comment]
First off, I want to say I understand what you are trying to do. I thank you for it. It's found some issues in the way I wrote the code that do need to be fixed. 

I also understand if you feel like I'm now expecting perfection (or at least some other level of coding) out of you then the current code exhibits and that is probably true. I don't like editing code once and then immediately turning around and editing it again. I think if we find a bug we should fix it the (mostly) right way then (I'm willing to give some leeway for expediency sake if the "real" right way is way more complicated and involved). 

So, let's take a step back. My chain of logic for why we should be using FlipPage::loginUrl and FlipPage::logoutURL is as follows:
A. FlipPage already needs to know the login and logout URLs in order to render the page header (Login button when there is no user logged in, Logout button otherwise).
B. Since FlipPage already has this variable (regardless of how it is calculated) why not simply use it on other pages so we don't have to recalculate or reacquire. While you are probably right and we probably won't change the url from profiles/login.php, why recalculate it?

Questions for you:
1. Do you disagree that FlipPage should be rendering the header and if so why?
2. Do you just disagree with the source of the loginUrl and logoutUrl variables (i.e. Settings) instead of calculating them or do you actually disagree with having them? If you disagree with having them then how would you suggest A work?
3. Can you see how I believe it would be easier to maintain if we used the same variable for the value everywhere?
4. Do you just think this isn't important because we probably aren't ever going to change the URLs?
5. Is there something I missed that you disagree with me about?

**Comment by russelltsherman** [comment]
answers..
1. i'm not clear to which header you are referring
2. my point was simply that our login endpoint will probably never change so it seems perfectly reasonable to use $this->profilesUrl with login.php appended
3. i agree that no matter what coding conventions we adopt.. consistency increases maintainability
4. exactly.
5. i don't think so

**Review by pboyd04 — APPROVED** [review]

**Pull request merged**